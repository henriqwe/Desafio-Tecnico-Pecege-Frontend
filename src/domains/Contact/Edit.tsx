import { IContact } from '&operations/queries/fetchContacts'
import { Table } from '&components/Table/Table'
import { Input } from '&components/Input/Input'

import { useForm } from 'react-hook-form'
import { useEffect, useState } from 'react'
import { useContacts } from '&contexts/contactsContext'
import { useNavigate, useParams } from 'react-router-dom'
import { Loading } from '&components/Loading/Loading'
import { routes } from '&utils/routes'

export function EditContact() {
  const navigate = useNavigate()
  const { id } = useParams()

  // const { contactsQuery, removeContact } = useContacts()
  function onSubmit(formData) {
    console.log({ formData })
  }
  const {
    handleSubmit,
    control,
    reset,
    formState: { errors }
  } = useForm()

  const { getContactById, contactsQuery } = useContacts()

  const contact = getContactById(id as string)

  if (contactsQuery.isLoading) {
    return <Loading />
  }
  if (!contact && !contactsQuery.isLoading) {
    throw new Error()
  }

  useEffect(() => {
    if (contact) {
      reset({ name: contact.name })
    }
  }, [contact])
  return (
    <div>
      <div onClick={() => navigate(routes.home.path)}>VOLTAR</div>
      <div>{contact?.name}</div>
      <div>
        <h1>Edição de contato</h1>
        <form onSubmit={handleSubmit(onSubmit)}>
          <Input control={control} fieldName="name" />
          <button>Salvar</button>
        </form>
      </div>
    </div>
  )
}

export const ErrorBoundary = () => {
  const navigate = useNavigate()

  return (
    <div>
      <div onClick={() => navigate(routes.home.path)}>VOLTAR</div>
      <h3>Some Error Boundary</h3>
    </div>
  )
}
