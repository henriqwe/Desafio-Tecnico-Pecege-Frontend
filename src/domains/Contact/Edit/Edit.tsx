import { useContacts } from '&contexts/contactsContext'
import { useNavigate, useParams } from 'react-router-dom'
import { Loading } from '&components/Loading/Loading'
import { routes } from '&utils/routes'
import { ContactForm } from '&components/ContactForm/ContactForm'
import { useQuery } from '@tanstack/react-query'
import { notification } from '&utils/notification'

export function EditContact() {
  const { id } = useParams()
  const { fetchContact } = useContacts()
  const navigate = useNavigate()

  const { data: contact, isLoading } = useQuery({
    queryKey: ['contact'],
    queryFn: () => fetchContact(id!),
    onError: (error: Error) => {
      notification(error.message, 'error')
      navigate(routes.home.path)
    },
    refetchOnWindowFocus: false,
    retry: false
  })

  if (isLoading || !contact) {
    return <Loading />
  }

  return <ContactForm contact={contact} />
}
