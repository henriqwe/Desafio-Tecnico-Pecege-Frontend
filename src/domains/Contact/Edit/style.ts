import styled from 'styled-components'

export const LoadingContainer = styled.div`
  display: flex;
  height: 100vh;
  flex: 1;
  flex-direction: column;
  gap: 3rem;
  align-items: center;
  justify-content: center;
`
