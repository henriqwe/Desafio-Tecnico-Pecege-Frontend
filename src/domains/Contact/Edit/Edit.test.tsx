import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { EditContact } from './Edit'
import { initialize } from '@googlemaps/jest-mocks'
import { TestProviders } from '&utils/tests/TestProviders'

jest.mock('@tanstack/react-query', () => ({
  ...jest.requireActual('@tanstack/react-query'),
  useQueryClient: () => ({
    setQueryData: jest.fn(),
    getQueryData: jest.fn()
  })
}))

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useParams: () => ({
    id: 3
  })
}))

beforeEach(() => {
  initialize()
})

describe('EditContact', () => {
  it('should render the contact list correctly', async () => {
    const { container, findByTestId } = render(
      <TestProviders>
        <EditContact />
      </TestProviders>
    )
    const name = (await findByTestId('input-name')) as HTMLInputElement
    expect(container).toMatchSnapshot()
    expect(name.value as string).toMatchSnapshot('Clementine Bauch')
  })
})
