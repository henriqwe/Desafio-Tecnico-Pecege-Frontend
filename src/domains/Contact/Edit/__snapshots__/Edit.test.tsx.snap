// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditContact should render the contact list correctly 1`] = `
<div>
  <section
    class="sc-dkjKgF hyIkKF"
  >
    <h1>
      Edição de contato
    </h1>
    <p>
      Clementine Bauch
    </p>
    <div
      class="sc-bcPKhP kmvBoQ"
    >
      <div
        class="sc-grXZZQ cyEMIS"
      >
        <div
          style="top: 19px; left: calc(-50% + 1.7rem); right: calc(50% + 1.7rem); position: absolute; background-color: rgb(212, 211, 209);"
        >
          <span
            style="border-top-style: solid; border-top-width: 2px; transition: all 0.2s linear; width: 100%; display: none; border-color: #18a2f3;"
          />
        </div>
        <div
          style="display: flex; margin: 0px auto 8px auto; align-items: center; justify-content: center; background-color: rgb(24, 162, 243); width: 2.5rem; height: 2.5rem; border-radius: 50%; transition: all 0.2s linear; transition-delay: 0.2s; position: relative;"
        >
          <div
            style="width: 1.5rem; height: 1.5rem; z-index: 1; color: rgb(24, 162, 243);"
          >
            <svg
              aria-hidden="true"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.75 6a3.75 3.75 0 11-7.5 0 3.75 3.75 0 017.5 0zM4.501 20.118a7.5 7.5 0 0114.998 0A17.933 17.933 0 0112 21.75c-2.676 0-5.216-.584-7.499-1.632z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
          <div
            style="background-color: rgb(255, 255, 255); border-radius: 50%; transition: all 0.2s linear; transition-delay: 0.2s; width: 2rem; height: 2rem; position: absolute;"
          />
        </div>
        <p
          style="text-align: center; font-size: 12px; transition: all 0.2s linear; transition-delay: 0.2s; color: rgb(24, 162, 243);"
        >
          Dados pessoais
        </p>
      </div>
      <div
        class="sc-grXZZQ cyEMIS"
      >
        <div
          style="top: 19px; left: calc(-50% + 1.7rem); right: calc(50% + 1.7rem); position: absolute; background-color: rgb(212, 211, 209);"
        >
          <span
            style="border-top-style: solid; border-top-width: 2px; transition: all 0.2s linear; width: 0px; display: block; border-color: rgb(212, 211, 209);"
          />
        </div>
        <div
          style="display: flex; margin: 0px auto 8px auto; align-items: center; justify-content: center; background-color: rgb(212, 211, 209); width: 2.5rem; height: 2.5rem; border-radius: 50%; transition: all 0.2s linear; transition-delay: 0.2s; position: relative;"
        >
          <div
            style="width: 1.5rem; height: 1.5rem; z-index: 1; color: rgb(75, 75, 75);"
          >
            <svg
              aria-hidden="true"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 10.5a3 3 0 11-6 0 3 3 0 016 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M19.5 10.5c0 7.142-7.5 11.25-7.5 11.25S4.5 17.642 4.5 10.5a7.5 7.5 0 1115 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
          <div
            style="background-color: rgb(255, 255, 255); border-radius: 50%; transition: all 0.2s linear; transition-delay: 0.2s; width: 0rem; height: 0rem; position: absolute;"
          />
        </div>
        <p
          style="text-align: center; font-size: 12px; transition: all 0.2s linear; transition-delay: 0.2s; color: rgb(176, 176, 176);"
        >
          Endereço
        </p>
      </div>
      <div
        class="sc-grXZZQ cyEMIS"
      >
        <div
          style="top: 19px; left: calc(-50% + 1.7rem); right: calc(50% + 1.7rem); position: absolute; background-color: rgb(212, 211, 209);"
        >
          <span
            style="border-top-style: solid; border-top-width: 2px; transition: all 0.2s linear; width: 0px; display: block; border-color: rgb(212, 211, 209);"
          />
        </div>
        <div
          style="display: flex; margin: 0px auto 8px auto; align-items: center; justify-content: center; background-color: rgb(212, 211, 209); width: 2.5rem; height: 2.5rem; border-radius: 50%; transition: all 0.2s linear; transition-delay: 0.2s; position: relative;"
        >
          <div
            style="width: 1.5rem; height: 1.5rem; z-index: 1; color: rgb(75, 75, 75);"
          >
            <svg
              aria-hidden="true"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.5 21v-7.5a.75.75 0 01.75-.75h3a.75.75 0 01.75.75V21m-4.5 0H2.36m11.14 0H18m0 0h3.64m-1.39 0V9.349m-16.5 11.65V9.35m0 0a3.001 3.001 0 003.75-.615A2.993 2.993 0 009.75 9.75c.896 0 1.7-.393 2.25-1.016a2.993 2.993 0 002.25 1.016c.896 0 1.7-.393 2.25-1.016a3.001 3.001 0 003.75.614m-16.5 0a3.004 3.004 0 01-.621-4.72L4.318 3.44A1.5 1.5 0 015.378 3h13.243a1.5 1.5 0 011.06.44l1.19 1.189a3 3 0 01-.621 4.72m-13.5 8.65h3.75a.75.75 0 00.75-.75V13.5a.75.75 0 00-.75-.75H6.75a.75.75 0 00-.75.75v3.75c0 .415.336.75.75.75z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
          <div
            style="background-color: rgb(255, 255, 255); border-radius: 50%; transition: all 0.2s linear; transition-delay: 0.2s; width: 0rem; height: 0rem; position: absolute;"
          />
        </div>
        <p
          style="text-align: center; font-size: 12px; transition: all 0.2s linear; transition-delay: 0.2s; color: rgb(176, 176, 176);"
        >
          Empresa
        </p>
      </div>
    </div>
    <form
      class="sc-eDnVMP jYoMzt"
    >
      <div
        class="sc-jSwlEQ bUSzpO"
      >
        <div>
          <div
            class="sc-fnOeiS ccaUeN"
          >
            <input
              autocomplete="off"
              class="sc-fEyylQ gdwRnj"
              data-testid="input-name"
              id="name"
              isinvalid="0"
              label="Nome *"
              placeholder="Clementine Bauch"
              type="text"
              value="Clementine Bauch"
            />
            <span
              class="sc-bjEwCx hbtnzl"
            >
               
            </span>
          </div>
          <label
            class="sc-ksJisA cxqOvz"
          />
        </div>
        <div>
          <div
            class="sc-fnOeiS ccaUeN"
          >
            <input
              autocomplete="off"
              class="sc-fEyylQ bAtXRS"
              data-testid="input-email"
              id="email"
              isinvalid="0"
              label="Email *"
              placeholder="Nathan@yesenia.net"
              type="text"
              value="Nathan@yesenia.net"
            />
            <span
              class="sc-bjEwCx hbtnzl"
            >
               
            </span>
          </div>
          <label
            class="sc-ksJisA cxqOvz"
          />
        </div>
        <div>
          <div
            class="sc-fnOeiS ccaUeN"
          >
            <input
              autocomplete="off"
              class="sc-fEyylQ hCaZTA"
              data-testid="input-phone"
              id="phone"
              isinvalid="0"
              label="Telefone *"
              placeholder="1-463-123-4447"
              type="text"
              value="1-463-123-4447"
            />
            <span
              class="sc-bjEwCx hbtnzl"
            >
               
            </span>
          </div>
          <label
            class="sc-ksJisA cxqOvz"
          />
        </div>
        <div>
          <div
            class="sc-fnOeiS ccaUeN"
          >
            <input
              autocomplete="off"
              class="sc-fEyylQ bXrllK"
              data-testid="input-username"
              id="username"
              isinvalid="0"
              label="Nome de usuário"
              placeholder="Samantha"
              type="text"
              value="Samantha"
            />
            <span
              class="sc-bjEwCx hbtnzl"
            >
               
            </span>
          </div>
          <label
            class="sc-ksJisA cxqOvz"
          />
        </div>
        <div>
          <div
            class="sc-fnOeiS ccaUeN"
          >
            <input
              autocomplete="off"
              class="sc-fEyylQ ZCCvb"
              data-testid="input-website"
              id="website"
              isinvalid="0"
              label="Website"
              placeholder="ramiro.info"
              type="text"
              value="ramiro.info"
            />
            <span
              class="sc-bjEwCx hbtnzl"
            >
               
            </span>
          </div>
          <label
            class="sc-ksJisA cxqOvz"
          />
        </div>
      </div>
      <div
        class="sc-hLclGa gqGEjV"
      >
        <button
          class="sc-eLSjS icvQtD"
          data-testid="button-cancel"
          type="button"
          variant="secondary"
        >
          <div
            class="sc-bqOBKd jmLTaX"
          >
            <svg
              aria-hidden="true"
              class="sc-gKHVLF hhjIlM"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 18L18 6M6 6l12 12"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <span>
              Cancelar
            </span>
          </div>
        </button>
        <button
          class="sc-eLSjS eWAJfw"
          data-testid="button-ok"
          type="submit"
          variant="primary"
        >
          <div
            class="sc-bqOBKd jmLTaX"
          >
            <span>
              Avançar
            </span>
            <svg
              aria-hidden="true"
              class="sc-ftLKQv hNvWEW"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </button>
      </div>
    </form>
  </section>
</div>
`;

exports[`EditContact should render the contact list correctly: Clementine Bauch 1`] = `"Clementine Bauch"`;
